sql-plugin/src/main/311+-all/scala/com/nvidia/spark/rapids/shims/v2/GpuRowBasedScalaUDF.scala

```diff
@@ -33,7 +33,8 @@ case class GpuRowBasedScalaUDF(
     outputEncoder: Option[ExpressionEncoder[_]],
     udfName: Option[String],
     nullable: Boolean,
-    udfDeterministic: Boolean)
+    udfDeterministic: Boolean,
+    var runTime: GpuMetric = NoopMetric)
```

Comment:
I'm a bit nervous about this approach.  Plan nodes get copied around and compared all the time, and the use of `var` in a case class is sort of an anti-pattern.  Also metrics are typically lazily created all over the code, yet we're forcing them to be evaluated up-front here.

Plans are supposed to be immutable, so typically when we want to change something in the plan we end up re-writing the plan with updated nodes.  I'm wondering if that's what we should be doing here, i.e.: rewriting the plan with an updated metric parameter which is not variable and is either a metric as above or possibly a function argument that provides an optional metric which would allow the metric creation to remain lazy.

Reply:
Oh, yes, I think you are right. I need more consideration for this.